tests/test_control_curves.py

```diff
@@ -516,3 +522,117 @@ def test_json(self, simple_storage_model):
         np.testing.assert_allclose(parameter.values, [[200, 100], [10, 5], [0, -10]])
         assert parameter.control_curves == [control_curve1, control_curve2]
         assert parameter.storage_node is model.nodes["Storage"]
+
+
+class TestWeightedAverageControlCurve:
+    def test_constant(self, three_storage_model):
+        m = three_storage_model
+
+        m.nodes["Storage 0"].max_volume = 16.0
+        curve0 = ConstantParameter(three_storage_model, 0.25)
+        curve1 = ConstantParameter(three_storage_model, 0.7)
+
+        storages = [m.nodes["Storage 0"], m.nodes["Storage 1"]]
+
+        profiles = [curve0, curve1]
+
+        agg_curves = WeightedAverageProfileParameter(m, storages, profiles)
+
+        m.setup()
+
+        assert_array_almost_equal(agg_curves.get_daily_values(), np.full(366, 0.5))
+
+    def test_monthly_and_constant(self, three_storage_model):
+        m = three_storage_model
+
+        m.nodes["Storage 0"].max_volume = 16.0
+        profile_vals = [
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+        ]
+        curve0 = MonthlyProfileParameter(three_storage_model, profile_vals)
+        curve1 = ConstantParameter(three_storage_model, 0.7)
+
+        storages = [m.nodes["Storage 0"], m.nodes["Storage 1"]]
+
+        profiles = [curve0, curve1]
+
+        agg_curves = WeightedAverageProfileParameter(m, storages, profiles)
+
+        m.setup()
+
+        assert_array_almost_equal(agg_curves.get_daily_values(), np.full(366, 0.5))
+
+    def test_monthly_and_daily(self, three_storage_model):
+        m = three_storage_model
+
+        m.nodes["Storage 0"].max_volume = 16.0
+        profile_vals = [
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+        ]
+        curve0 = MonthlyProfileParameter(three_storage_model, profile_vals)
+        curve1 = DailyProfileParameter(three_storage_model, np.full(366, 0.7))
+
+        storages = [m.nodes["Storage 0"], m.nodes["Storage 1"]]
+
+        profiles = [curve0, curve1]
+
+        agg_curves = WeightedAverageProfileParameter(m, storages, profiles)
+
+        m.setup()
+
+        assert_array_almost_equal(agg_curves.get_daily_values(), np.full(366, 0.5))
+
+    def test_varying_curve(self, three_storage_model):
```

Comment:
Could this test the interp_day argument on the `MonthlyProfileParameter`?

Reply:
done